### ISSUE TITLE
Single machine/wm deploy instead of Kubernetes

### ISSUE THREAD

**Issue by brillout** [kubesail/deploy-to-kube#25]
deploy-node-app looks neat, I was thinking of doing something like that in the context of [Reframe](https://github.com/reframejs/reframe). I might as well just use what your building. Cool.

Kubernetes is great to scale up, but for prototypes and small/medium traffic apps, it's overkill. 

What do you think of supporting single machine/vm deployments with Nix? That'd be easier for smaller projects.

If my app has lots of traffic, chances are that I have the money to hire a devOp and I don't need deploy-node-app anymore.

It's also nice to have a similar env in prod than in dev (a single machine/vm deployment is closer to my dev machine than a kube deployment).

**Comment by pastudan** [comment]
Hey @brillout, thanks for checking out deploy-node-app!

I agree that for small projects, Kubernetes can seem like overkill. Deploy node app was actually created to try and solve that.

I will admit we are biased toward Kubernetes... @erulabs and I started KubeSail to help lower the barrier of entry of Kubernetes for the average developer. We think EKS, GKE, etc got it wrong by handing you an entire cluster, so instead we give you a namespace on our managed cluster, with tools like deploy-node-app to help get your apps deployed on there. Once you have the need to run your own cluster, then its just a matter of moving over the configs rather than doing a big migration.

In our careers, we've worked at places that started as side projects and have grown to handle massive amounts of traffic (most notably, Imgur). Orgs like these would have benefitted greatly from something like KubeSail, where things are set up correctly from the beginning, and can scale to handle millions of users without having to transition between platforms.

We are also trying to close the gap between dev & prod. I recognize that installing MiniKube as a pre-requisite for developing is not ideal, so we are developing some tools that let you develop against remote dependencies. As an example, I could develop my Node.js app on my laptop, but it would proxy the Postgres and Redis traffic back to my Kube namespace. Something akin to https://github.com/txn2/kubefwd

Sorry for the long-winded answer, but I hope this helps you understand why the focus of this project will stay geared toward Kubernetes.

If there's something that could be made simpler, either with deploy-node-app or KubeSail, I'd love to hear your feedback! I certainly still agree that Kubernetes has some major hurdles to overcome to become as simple as launching on a single VM, and those hurdles are what I'm excited to fix!

Also, thanks for your work on Reframe, it looks very cool! I'll have to give it a try for my next project.
